indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -1263,16 +1047,19 @@ public static class Builder
     private TuningConfig tuningConfig;
     @Nullable
     private Map<String, Object> context;
+    private CompactionRunner compactionRunner;
 
     public Builder(
         String dataSource,
         SegmentCacheManagerFactory segmentCacheManagerFactory,
-        RetryPolicyFactory retryPolicyFactory
+        RetryPolicyFactory retryPolicyFactory,
+        CompactionRunner compactionRunner
```

Comment:
Not needed in the constructor here as you have already added a separate builder method to set the `compactionRunner`. Fields passed to the constructor of the builder must be final.